discovery-provider/src/models.py

```diff
@@ -13,8 +18,46 @@
     Enum,
     PrimaryKeyConstraint,
 )
+from src.model_validator import ModelValidator
 
 Base = declarative_base()
+logger = logging.getLogger(__name__)
+
+def validate_field_helper(field, value, model):
+    # TODO: need to write custom validator for these datetime fields as jsonschema
+    # validates datetime in format 2018-11-13T20:20:39+00:00, not a format we use
+    # also not totally necessary as these fields are created server side
+    if field in ('created_at', 'updated_at'):
+        return value
+
+    to_validate = {
+        field: value
+    }
+
+    try:
+        ModelValidator.validate(to_validate=to_validate, model=model, field=field)
+    except ValidationError as e:
+        field_type = ModelValidator.get_properties_field_type(model, field)
+        default = ModelValidator.get_properties_field_default(model, field)
+
+        if not field_type or field_type == 'object':
```

Comment:
will the default json schema set type object to sql null?

Reply:
yes! if the default is specified as none in the schema, the db value will be sql null. for fields that are object types, we have to set the null to sql null() explicitly